src/parser.ts

```diff
@@ -10,7 +10,7 @@ import BigNumber from "bignumber.js";
 export interface Payment {
   receiver: string;
   amount: BigNumber;
-  tokenAddress: string;
+  tokenAddress: string | null;
```

Comment:
We definitely needed this!

Reply:
Isn't this the same as 

```js
tokenAddress?: string
```

Reply:
Nope that would mean string | undefined but null is not the same as undefined
We should change our tsconfig to include strictNullChecks. It will then be a compiler error if we forget to Nullcheck / undefined check something.